plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryResultPageSource.java

```diff
@@ -165,6 +167,12 @@ else if (javaType == long.class) {
                 else if (type.equals(INTEGER)) {
                     type.writeLong(output, ((Number) value).intValue());
                 }
+                else if (type instanceof DecimalType) {
+                    verify(isShortDecimal(type));
```

Comment:
An explicit message when verify fails (even though we don't expect it to) would be useful.